packages/browser-wallet/src/background/confirmation.ts

```diff
@@ -0,0 +1,136 @@
+
+async function monitorCredentialStatus(
+    jsonRpcUrl: string,
+    { deploymentHash, ...info }: PendingWalletCredential,
+    signal: AbortSignal
+) {
+    const client = new JsonRpcClient(new HttpProvider(jsonRpcUrl, fetch));
+
+    setTimeout(async function loop() {
+        if (signal.aborted) {
+            return;
+        }
+        let repeat = true;
+        try {
+            const response = await client.getTransactionStatus(deploymentHash);
+            if (response?.status === TransactionStatusEnum.Finalized) {
+                const isSuccessful = Object.values(response?.outcomes || {}).some(
+                    (outcome) => outcome.result.outcome === 'success'
+                );
+                await updateCredentials([
+                    {
+                        ...info,
+                        status: isSuccessful ? CreationStatus.Confirmed : CreationStatus.Rejected,
+                    },
+                ]);
+                repeat = false;
+            }
+        } finally {
+            if (repeat) {
+                setTimeout(loop, updateInterval);
+            }
+        }
+    }, 0);
+}
+
+/**
+ * Start checks on all pending credentials on the current network.
+ */
+async function startMonitoringCredentialStatus(signal: AbortSignal) {
+    const network = await storedCurrentNetwork.get();
+    if (network) {
+        const url = network.jsonRpcUrl;
+        const creds = await storedCredentials.get(network.genesisHash);
+        if (url && creds) {
+            creds.filter(isPendingCred).forEach((cred) => monitorCredentialStatus(url, cred, signal));
+        }
+    }
+}
+
+/**
+ * Continously checks whether pending identities have been confirmed or rejected.
+ */
+async function monitorIdentityStatus({ location, ...identity }: PendingIdentity, signal: AbortSignal) {
+    setTimeout(async function loop() {
+        if (signal.aborted) {
+            return;
+        }
+        let repeat = true;
+        try {
+            const response = (await (await fetch(location)).json()) as IdentityTokenContainer;
+            if (response.status === IdentityProviderIdentityStatus.Error) {
+                updateIdentities([
+                    {
+                        ...identity,
+                        status: CreationStatus.Rejected,
+                        error: response.detail,
+                    },
+                ]);
+                repeat = false;
+            } else if (response.status === IdentityProviderIdentityStatus.Done) {
+                updateIdentities([
+                    {
+                        ...identity,
+                        status: CreationStatus.Confirmed,
+                        idObject: response.token.identityObject,
+                    },
+                ]);
+                repeat = false;
+            }
+        } finally {
+            if (repeat) {
+                setTimeout(loop, updateInterval);
+            }
+        }
+    }, 0);
```

Comment:
It feels hacky to set a 0 time timeout. Could you just define the `loop` function and then run it on the next line instead?

Reply:
Yes.